lib/board.rb

```diff
@@ -6,9 +6,82 @@ class Board
   # default board size = 4
   def initialize(board_size = 4)
     # @board_size = board_size
-    # @cells =
+    @cells = {
+      "A1" => Cell.new("A1"),
+      "A2" => Cell.new("A2"),
+      "A3" => Cell.new("A3"),
+      "A4" => Cell.new("A4"),
+      "B1" => Cell.new("B1"),
+      "B2" => Cell.new("B2"),
+      "B3" => Cell.new("B3"),
+      "B4" => Cell.new("B4"),
+      "C1" => Cell.new("C1"),
+      "C2" => Cell.new("C2"),
+      "C3" => Cell.new("C3"),
+      "C4" => Cell.new("C4"),
+      "D1" => Cell.new("D1"),
+      "D2" => Cell.new("D2"),
+      "D3" => Cell.new("D3"),
+      "D4" => Cell.new("D4")
+     }
   end
 
+  def valid_coordinate?(cell)
+    @cells.include?(cell)
```

Comment:
```
def valid_coordinate?(coordinate)
  cells.keys.include?(coordinate)
end
```

@cells is a hash, so we need to make sure that the coordinates is one of the cell keys.